pyigm/cgm/cgmsurvey.py

```diff
@@ -149,7 +156,133 @@ def nsys(self):
+                                          flag_N=igm_comp[comp]['flag_N'],
+                                          sig_logN=np.array([igm_comp[comp]['sig_logN']]*2)))
+            # Now sum em up
+            if len(comps) == 0:
+                flagNs.append(0)
+                Ns.append(0.)
+                sigNs.append(np.array([0.]*2))
+                continue
+            obj = dict(flag_N=comps[0]['flag_N'], logN=comps[0]['logN'], sig_logN=comps[0]['sig_logN'])
+            for comp in comps[1:]:
+                if comp['flag_N'] != 0:
+                    obj['flag_N'], obj['logN'], obj['sig_logN'] = ltaa.sum_logN(obj, comp)
+            # Save
+            flagNs.append(obj['flag_N'])
+            Ns.append(obj['logN'])
+            sigNs.append(obj['sig_logN'])
+        # Add to Table
+        self._data.add_column(Column(flagNs, name='flag_N_{:s}'.format(ion)))
+        self._data.add_column(Column(Ns, name='logN_{:s}'.format(ion)))
+        self._data.add_column(Column(sigNs, name='sig_logN_{:s}'.format(ion)))
+
+    def build_sys_from_dict(self, sys_name, append=False, llist=None, **kwargs):
+        tdict = self._dict[sys_name]
+        cgmsys = CGMAbsSys.from_dict(tdict, chk_vel=False, chk_sep=False, chk_data=False,
+                                 use_coord=True, use_angrho=True,
+                                     linelist=llist, **kwargs)
+        if append:
+            self.cgm_abs.append(cgmsys)
+        # Return
+        return cgmsys
+
+
+    def build_systems_from_dict(self, **kwargs):
+        for key in self._dict.keys():
+            cgmsys = self.build_sys_from_dict(key, **kwargs)
+            self.cgm_abs.append(cgmsys)
+
+    def data_from_dict(self):
+        # Table columns
+        key0 = list(self._dict.keys())[0]
+        tab_clms = list(self._dict[key0].keys())
+        # Handle RA/DEC
+        IGM_RA = []
+        IGM_DEC = []
+        zabs = []
+        for key in self._dict.keys():
+            IGM_RA.append(self._dict[key]['igm_sys']['RA'])
+            IGM_DEC.append(self._dict[key]['igm_sys']['DEC'])
+            zabs.append(self._dict[key]['igm_sys']['zabs'])
+        self._data.add_column(Column(IGM_RA, name='RA_IGM'))
+        self._data.add_column(Column(IGM_DEC, name='DEC_IGM'))
+        self._data.add_column(Column(zabs, name='zabs'))
+        # Remove unwanted ones
+        rmv_keys = ['CreationDate', 'cosmo', 'ebv', 'user', 'Refs', 'igm_sys', 'galaxy']
+        for rkey in rmv_keys:
+            if rkey in tab_clms:
+                tab_clms.remove(rkey)
+        # Build it
+        for tclm in tab_clms:
+            values = []
+            for key in self._dict.keys():
+                values.append(self._dict[key][tclm])
+            # Add column
+            clm = Column(values, name=tclm)
+            self._data.add_column(clm)
+
+    def to_json(self, outfile, overwrite=True):
+        """ Generates a gzipped JSON file of the survey
```

Comment:
The JSON file won't be gzipped unless the user does it after running this method, right?

Reply:
agreed.  Have added doc strings elsewhere too.